kbs/src/api_server.rs

```diff
@@ -0,0 +1,317 @@
+                    .await
+                    .map_err(|_| Error::TokenNotFound)?;
+
+                let claims = core.token_verifier.verify(token).await?;
+
+                let claim_str = serde_json::to_string(&claims)?;
+                if !core
+                    .policy_engine
+                    .evaluate(resource_desc, &claim_str)
+                    .await?
+                {
+                    return Err(Error::PolicyDeny);
+                };
+
+                let resource_description = ResourceDesc::try_from(resource_desc)?;
+                let resource = core
+                    .resource_storage
+                    .get_secret_resource(resource_description)
+                    .await?;
+
+                let public_key = core.token_verifier.extract_tee_public_key(claims)?;
+                let jwe = jwe(public_key, resource).map_err(|e| Error::JweError { source: e })?;
+
+                let res = serde_json::to_string(&jwe)?;
+
+                Ok(HttpResponse::Ok()
+                    .content_type("application/json")
+                    .body(res))
+            } else if request.method() == Method::POST {
+                let resource_desc =
+                    sub_path
+                        .strip_prefix('/')
+                        .ok_or(Error::IllegalAccessedPath {
+                            path: end_point.clone(),
+                        })?;
+                let resource_description = ResourceDesc::try_from(resource_desc)?;
+                core.admin_auth.validate_auth(&request)?;
+                core.resource_storage
+                    .set_secret_resource(resource_description, &body)
+                    .await?;
+
+                Ok(HttpResponse::Ok().content_type("application/json").body(""))
+            } else {
+                Ok(HttpResponse::NotImplemented()
+                    .content_type("application/json")
+                    .body(""))
+            }
+        }
+        plugin_name => {
+            // Plugin calls needs to be authorized by the Token and policy
+            let token = core
+                .get_attestation_token(&request)
+                .await
+                .map_err(|_| Error::TokenNotFound)?;
+
+            let claims = core.token_verifier.verify(token).await?;
+
+            let claim_str = serde_json::to_string(&claims)?;
+
+            // TODO: add policy filter support for other plugins
+            if !core.policy_engine.evaluate(&end_point, &claim_str).await? {
+                return Err(Error::PolicyDeny);
+            }
+
+            let plugin = core
+                .plugin_manager
+                .get(plugin_name)
+                .ok_or(Error::PluginNotFound {
+                    plugin_name: plugin_name.to_string(),
+                })?;
+            let body = body.to_vec();
+            let response = plugin
+                .handle(body, query.into(), sub_path.into(), request.method())
```

Comment:
I think the plugin may/should want to encrypt the response payload. If so, the `handle()` should be able to access the public_key. You could pass it via parameter.

Reply:
Nice idea.

Reply:
`encrypted()` gives some flexibility to the plugin in terms of encrypting the payload. However, I can't think of a plugin that would want its payload to be sent back unencrypted. If we don't have a use case for that we could just assume that all plugins want its payload to be encrypted.

Reply:
There is one in my mind is to return the public cert whose confidentiality does not need to be protected. The request from guest is the evidence and the CSR of tee public key.